File: prob901/py/Solution.py
from typing import List

    

class StockSpanner:

    def __init__(self):
        self.xs = []
        self.ls = []

    def next(self, price: int) -> int:

        t = 1
        while len(self.xs) != 0 and self.xs[-1] <= price:
            t += self.ls.pop()
            self.xs.pop()
        self.ls.append(t)
        self.xs.append(price)
        return t

File: prob901/java/Solution.java
import java.util.Stack;

class StockSpanner {
    Stack<Integer> xs;
    Stack<Integer> ys;

    public StockSpanner() {
        xs = new Stack<>();
        ys = new Stack<>();
    }
    
    public int next(int price) {
        int t = 1;
        while (!xs.isEmpty() && xs.peek() <= price) {
            t += ys.pop();
            xs.pop();
        }
        xs.push(price);
        ys.push(t);
        return t;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */